rosidl_runtime_c/src/type_hash.c

```diff
@@ -0,0 +1,151 @@
+// Copyright 2023 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string.h>
+
+#include "rosidl_runtime_c/type_hash.h"
+
+#include "rcutils/error_handling.h"
+
+static const char RIHS01_PREFIX[] = "RIHS01_";
+// Hash representation is hex string, two characters per byte
+static const size_t RIHS_VERSION_IDX = 4;
+static const size_t RIHS_PREFIX_LEN = 7;
```

Comment:
Can this just be `sizeof(RIHS01_PREFIX)`?

Quick example, seems to imply yes: https://godbolt.org/z/W9hcnP9Ph

Reply:
I tried to do that  but MSVC doesn't like it - it  doesn't consider `sizeof` to be a const expression.

Reply:
But not according to godbolt's MSVC option here... maybe I was confused?

Reply:
ah, the issue was when I was trying to do like:

```
static const size_t THING1 = 1;
static const size_t  THING2 = 2;
static const size_t THING3 = THING1 + THING2;
```

which gcc thinks is fine,  but MSVC does not like. 

Sizeof would work here, though the standard we've created says that this number is locked in for all future versions

Reply:
Thanks for (re)investigating, I think this is fine as-is, it's a minor point.